juicer/workflow/workflow.py

```diff
@@ -1,27 +1,263 @@
+        if self.check_null_target_id_tasks() \
+                and \
+            self.check_null_source_id_tasks():
+
+            self.sorted_tasks = self.get_topological_sorted_tasks()
+        else:
+            raise AttributeError(
+                "Parameter '{}/{}' must be informed for task {}".format(
+                    self.WORKFLOW_GRAPH_SOURCE_ID_PARAM,
+                    self.WORKFLOW_GRAPH_TARGET_ID_PARAM,
+                    self.__class__))
+
+
+
+    def builds_initial_workflow_graph(self):
+        """ Builds a graph with the tasks """
+
+        for task in self.workflow_data['tasks']:
+
+            query_tahiti = self.get_ports_from_operation_tasks(
+                                    task.get('operation')['id'])
+
+            self.workflow_graph.add_node(
+                task.get('id'),
+                tahiti_indegree_required=query_tahiti['N_INPUT'],
+                tahiti_indegree_multiplicity=query_tahiti['M_INPUT'],
+                tahiti_outdegree_required=query_tahiti['N_OUTPUT'],
+                tahiti_outdegree_multiplicity=query_tahiti['M_OUTPUT'],
+                attr_dict=task)
+
+        for flow in self.workflow_data['flows']:
+            self.workflow_graph.add_edge(flow['source_id'], flow['target_id'],
                                 attr_dict=flow)
 
-            # Workflow dictionary
-            # self.workflow = workflow_data
-            # Topological sorted tasks according to their dependencies
-            # self.sorted_tasks = []
+        for nodes in self.workflow_graph.nodes():
+            self.workflow_graph.node[nodes]['in_degree'] = self.workflow_graph.\
+                in_degree(nodes)
 
-    # def set_workflow(self, workflow):
-    #     self.workflow = workflow
+            self.workflow_graph.node[nodes]['out_degree'] = self.workflow_graph.\
+                out_degree(nodes)
 
-    # def verify_workflow(self):
-    #     """
+        return self.workflow_graph
+
+
+    def check_in_degree_edges(self):
+        for nodes in self.workflow_graph.nodes():
+            if self.workflow_graph.node[nodes]['in_degree'] == \
+                    self.workflow_graph.node[nodes]['tahiti_indegree_required']:
+                pass
+            else:
+                raise AttributeError(
+                    "Parameter '{} in node {}' missing, must be informed for flow {}".
+                    format(
+                        self.WORKFLOW_GRAPH_TARGET_ID_PARAM,
+                        nodes,
+                        self.__class__))
+        return 1
+
+    def check_out_degree_edges(self):
+
+        for nodes in self.workflow_graph.nodes():
+            if self.workflow_graph.node[nodes]['out_degree'] == \
+                    self.workflow_graph.node[nodes]['tahiti_outdegree_required']:
+                pass
+            else:
+                raise AttributeError(
+
+                    "Parameter '{}' missing, must be informed for flow {}".format(
```

Comment:
It is not a parameter, it is a port. And is not a flow in the end, is an operation.